src/main/java/org/fregelang/plugin/idea/run/FregeCommandLineState.java

```diff
@@ -0,0 +1,213 @@
+package org.fregelang.plugin.idea.run;
+
+import com.intellij.execution.ExecutionException;
+import com.intellij.execution.configurations.CommandLineState;
+import com.intellij.execution.configurations.GeneralCommandLine;
+import com.intellij.execution.process.KillableColoredProcessHandler;
+import com.intellij.execution.process.ProcessHandler;
+import com.intellij.execution.runners.ExecutionEnvironment;
+import com.intellij.ide.highlighter.JavaClassFileType;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.roots.*;
+import com.intellij.openapi.roots.libraries.Library;
+import com.intellij.openapi.vfs.VirtualFile;
+import com.intellij.psi.search.FileTypeIndex;
+import com.intellij.psi.search.GlobalSearchScope;
+import com.intellij.util.indexing.FileBasedIndex;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+
+/**
+ * Represents a Command that gets executed for compiling and executing Frege Code
+ *
+ * @author rahulsom
+ */
+public class FregeCommandLineState extends CommandLineState {
+  private FregeRunConfiguration runConfiguration;
+
+  public FregeCommandLineState(FregeRunConfiguration runConfiguration, ExecutionEnvironment executionEnvironment) {
+    super(executionEnvironment);
+    this.runConfiguration = runConfiguration;
+  }
+
+  @NotNull
+  @Override
+  protected ProcessHandler startProcess() throws ExecutionException {
+    System.out.println("StartProcess");
+
+    final Project project = runConfiguration.getProject();
+    final ProjectRootManager projectRootManager = ProjectRootManager.getInstance(project);
+
+    final VirtualFile[] sourceRoots = projectRootManager.getContentSourceRoots();
+    final String sourcePath = getSourcePath(sourceRoots);
+    final String fregeClassPath = getClassPathSource();
+    final String projectOutPath = project.getBasePath() + "/out";
+    final String effectiveClassPath = fregeClassPath + ":" + projectOutPath;
+
+    /*
+     * Cross compile from frege to java
+     */
+    ProcessHandler fregec = compile(sourcePath, ".fr", file -> new String[]{
```

Comment:
I tried doing that. This is my groovy sample.

```groovy
@Grab(group='org.frege-lang', module='frege', version='3.23.370-g898bc8c')

import frege.compiler.Main
import frege.prelude.PreludeBase
import frege.runtime.Lambda
import frege.runtime.Runtime

println "Compile phase 1"
def var0 = ["-d", "../out","-sp", ".", "-j", "-nocp", "-greek", "demo.fr"]
Runtime.runMain(PreludeBase.TST.performUnsafe((Lambda)Main.Ĳ._mainƒd0fa0028.inst.apply(PreludeBase._toList(var0)).forced()));
println "Compile phase 2"
Runtime.runMain(PreludeBase.TST.performUnsafe((Lambda)Main.Ĳ._mainƒd0fa0028.inst.apply(PreludeBase._toList(var0)).forced()));
println "Done"
```

That causes this output 
```
Compile phase 1
Compile phase 2
Caught: java.util.concurrent.RejectedExecutionException: Task java.util.concurrent.FutureTask@79da1ec0 rejected from java.util.concurrent.ThreadPoolExecutor@19fb8826[Terminated, pool size = 0, active threads = 0, queued tasks = 0, completed tasks = 1]
... stacktrace
```

Calling `Main.main(...)` is not suitable because it causes a `System.exit`.

Is there a cleaner way of calling the compiler from java code?